### ISSUE TITLE
NullPointerException can result from line 163 of SecurityWrapperRequest.java:

### ISSUE THREAD

**Issue by mvkendall** [ESAPI/esapi-java-legacy#515]
Cookie n = new Cookie(name, value); // when name is null NPE is thrown

java.lang.NullPointerException
        at javax.servlet.http.Cookie.isToken(Cookie.java:514)
        at javax.servlet.http.Cookie.<init>(Cookie.java:135)

**Comment by xeno6696** [comment]
Would be lots faster if you'd fork the code and then add the null check, then submit a PR!  We need help on this project!

**Comment by kwwall** [comment]
The problem I have with this is what is the best way to handle this? Any exception that is thrown will have to be an _unchecked_ exception because we can't change public interfaces without potentially breaking code and the default `NullPointerException` that is thrown is probably the a reasonable unchecked exception type here, although not the best choice. Since ESAPI is not setting the cookies in this case, but rather just grabbing the Cookie: HTTP request header from the HTTP request, it's not really appropriate (IMO at least) to throw an `IllegalArgumentException` here.

Now this case definitely _could_ happen from an attack, so yes, we should to do something here. IMO throwing any sort of `Error` for an unchecked exception is just wrong, so it will have to be some sort of `RuntimeException`. ESAPI has 3 such exceptions: the base class `EnterpriseSecurityRuntimeException` and subclasses `EncryptionException` and `IntrusionException`. Unless one of you can think of how this might accidentally (and _commonly_) arise from a programming error, I think we should go with `IntrusionException`. As per the Javadoc for `IntrusionException`:

> An `IntrusionException` should be thrown anytime an error condition arises that is likely to be the result of an attack in progress.

Specifically, I think we should use the 3 arg CTOR here. I think that the 'logMessage' argument should be something like "HTTP request detected Cookie: header where cookie did not have a required name". (I don't think we should log the cookie _value_ because it potentially could be a sensitive value like a session ID, etc. that really ought not to be logged.) As for the 'userMessage', I'm open to suggestions there. And for the 'cause' argument, just pass in the NPE that is caught. But that's my $.02 on this if @mvkendall wants to do a PR for us.

Interested in hearing other's opinions about this though.

**Comment by kwwall** [comment]
I took another closer look at this. Seems to me that line 157, which is this:

`                String name = ESAPI.validator().getValidInput("Cookie name: " + c.getName(), c.getName(), "HTTPCookieName", sc.getIntProp("HttpUtilities.MaxHeaderNameSize"), true);`

is the root case of this problem. That last argument, set to `true`, probably should have been `false`. That last argument determines whether or not a null value or empty string for the input (in this case, the cookie name) is considered legal. I would argue that a null or empty cookie name should _not_ be considered legal, so we probably should have set that 5th argument to `false` rather than `true`. I'm pretty sure that that would fix it by causing a `ValidationException` to be thrown, which we already catch, meaning that it is not likely to break anyone's code. (We log the value, which I'm a bit concerned about [see my previous comment], but I suspect if we did a manual code review, we'd probably find a bunch of places like this where we log things when we maybe shouldn't. At least there probably should be a note in the Javadoc.)

So maybe this is the better way to fix it--just change the `true` to `false` on that line--instead of adding a catch for `NullPointerException` and then re-throwing it as an `IntrusionException` as I described in my previous comment.

Let's discuss.
